crates/dojo-indexer/src/indexer.rs

```diff
@@ -59,28 +62,22 @@ pub async fn start_indexer(
     // filter requested data by the events we process
     filter_by_processors(&mut filter, processors);
 
-    let data_stream = stream
-        .request_data({
-            Filter {
-                header: Some(HeaderFilter { weak: false }),
-                transactions: vec![],
-                events: vec![],
-                messages: vec![],
-                state_update: None,
-            }
-        })
-        .await?;
-    futures::pin_mut!(data_stream);
+    // TODO: should set starting block.
+    let starting_configuration = Configuration::<Filter>::default()
+        .with_finality(DataFinality::DataStatusAccepted)
+        .with_filter(|filter| filter.with_header(HeaderFilter { weak: false }));
+    stream_client.send(starting_configuration).await?;
```

Comment:
What looks strange in the old code is that after creating the `filter`, it was never touched again and a new (empty) filter was used instead.